diff --git a/vendor/magento/module-banner/view/frontend/web/js/model/banner.js b/vendor/magento/module-banner/view/frontend/web/js/model/banner.js
index 5dced94763..32cc02903f 100644
--- a/vendor/magento/module-banner/view/frontend/web/js/model/banner.js
+++ b/vendor/magento/module-banner/view/frontend/web/js/model/banner.js
@@ -26,16 +26,32 @@ define([
         invalidateCacheBySessionTimeOut = function () {
             var cacheEol = new Date($.localStorage.get('mage-banners-cache-timeout')),
                 dateTo = new Date(Date.now() + options.cacheTtl),
+                cartDataId = 0,
                 globalStoreId = $.cookieStorage.get('store') || 'default';

             if ($.localStorage.get('mage-banners-storeId') === null) {
                 $.localStorage.set('mage-banners-storeId', globalStoreId);
             }

-            if (cacheEol < new Date() || $.localStorage.get('mage-banners-storeId') !== globalStoreId) {
+            if ($.localStorage.get('mage-banners-cartDataId') === null && cartDataId > 0) {
+                $.localStorage.set('mage-banners-cartDataId', cartDataId);
+            }
+
+            if ($.localStorage.get('mage-cache-storage') !== null &&
+                $.localStorage.get('mage-cache-storage').hasOwnProperty('cart')) {
+                cartDataId = $.localStorage.get('mage-cache-storage').cart['data_id'];
+            }
+
+            if (cacheEol < new Date() ||
+                $.localStorage.get('mage-banners-storeId') !== globalStoreId ||
+                $.localStorage.get('mage-banners-cartDataId') !== cartDataId) {
                 storage.removeAll();
                 $.localStorage.set('mage-banners-cache-timeout', dateTo);
                 $.localStorage.set('mage-banners-storeId', globalStoreId);
+
+                if (cartDataId > 0) {
+                    $.localStorage.set('mage-banners-cartDataId', cartDataId);
+                }
             }
         },
         dataProvider = {
